File: tests/test-minecraft-osp/src/main/java/pize/tests/voxelgame/main/chat/MessageSource.java
package pize.tests.voxelgame.main.chat;

public class MessageSource{
    
    private final MessageSources source;
    
    public MessageSource(MessageSources source){
        this.source = source;
    }
    
    public MessageSources getSource(){
        return source;
    }
    
    
    public boolean isPlayer(){
        return source == MessageSources.PLAYER;
    }
    
    public boolean isServer(){
        return source == MessageSources.SERVER;
    }
    
    
    public MessageSourcePlayer asPlayer(){
        return (MessageSourcePlayer) this;
    }
    
}

File: tests/test-minecraft-osp/src/main/kotlin/pize/tests/voxelgame/main/chat/MessageSource.kt
package pize.tests.voxelgame.main.chat

open class MessageSource(val source: MessageSources) {

    val isPlayer: Boolean
        get() = source == MessageSources.PLAYER
    val isServer: Boolean
        get() = source == MessageSources.SERVER

    fun asPlayer(): MessageSourcePlayer {
        return this as MessageSourcePlayer
    }
}
